src/argilla/client/feedback/training/frameworks/sentence_transformers.py

```diff
@@ -0,0 +1,210 @@
+
+    def init_model(self) -> None:
+        """
+        Initializes a model.
+        """
+        if "model_name_or_path" in self.model_kwargs:
+            self.model_kwargs.pop("model_name_or_path")
+        if "model_name" in self.model_kwargs:
+            self.model_kwargs.pop("model_name")
+
+        if "train_objectives" in self.trainer_kwargs:
+            self.trainer_kwargs.pop("train_objectives")
+        if "train_dataloader" in self.trainer_kwargs:
+            self.trainer_kwargs.pop("train_dataloader")
+
+        self._trainer = self._trainer_cls(self._model, **self.model_kwargs)
+
+    def update_config(self, **kwargs) -> None:
+        """
+        Updates the configuration of the trainer, but the parameters depend on the trainer.subclass.
+        """
+        self.model_kwargs.update(filter_allowed_args(self._trainer_cls.__init__, **kwargs))
+        self.trainer_kwargs.update(filter_allowed_args(self._trainer_cls.fit, **kwargs))
+
+        if "batch_size" in kwargs:
+            self.dataloader_kwargs["batch_size"] = kwargs["batch_size"]
+
+        if "model" in self.model_kwargs:
+            self._model = self.model_kwargs["model"]
+            # self._model = self.model_kwargs["model_name_or_path"]
+
+    def train(self, output_dir: Optional[str] = None) -> None:
+        """
+        Trains the model.
+        """
+        from torch.utils.data import DataLoader
+
+        self.init_model()
+
+        if self._dataset_type:
+            dataloader = DataLoader(
+                dataset=self._dataset_type(self._train_dataset), batch_size=self.dataloader_kwargs["batch_size"]
+            )
+        else:
+            dataloader = DataLoader(dataset=self._train_dataset, batch_size=self.dataloader_kwargs["batch_size"])
+
+        train_loss = self._loss_cls(self._trainer)
+
+        if self._cross_encoder:
+            train_objectives = dataloader
+        else:
+            train_objectives = [(dataloader, train_loss)]
+
+        self._trainer.fit(train_objectives, **self.trainer_kwargs)
+        # NOTE: We haven't used the evaluator yet
+
+        if output_dir is not None:
+            self.save(output_dir)
+
+    def predict(self, text: Union[List[str], str], as_argilla_records: bool = True, **kwargs):
+        """
+        Predicts the label of the text.
+        """
+        raise NotImplementedError("work in progress")
+
+    def save(self, output_dir: str) -> None:
+        """
+        Saves the model to the specified path.
+        """
+        if self._cross_encoder:
+            self._trainer.save(output_dir)
+        else:
+            # For the moment just save the in the simplest form, creating the model card or the
+            # dataset for example should be done taking the extra information from the argilla
+            # dataset instead of the defaults
+            self._trainer.save(output_dir, model_name=None, create_model_card=False, train_datasets=None)
+
+    def __repr__(self) -> str:
```

Comment:
why did you add this __repr__? One of the subclasses defines this already and outlines the arg and kwargs for the training config.

Reply:
I thought I would need to adjust the subclass default a bit for the case like you did in the `openai.py` or `trl.py` cases,
and I let it there while I took a look, but I really don't now if it needs more info. I will remove it to let the subclass one.